packages/storage-postgres/src/impl/postgres-storage.ts

```diff
@@ -0,0 +1,52 @@
+import { Storage, Logger, Repositories } from 'steveo';
+import { PrismaClient } from '@prisma/client';
+import { WorkflowStateRepositoryPostgres } from '../repo/workflow-postgres-repo';
+import { PostgresStorageConfig } from './postgres-config';
+
+/**
+ *
+ */
+class PostgresStorage extends Storage {
+  prisma: PrismaClient;
+
+  constructor(private config: PostgresStorageConfig, private logger: Logger) {
+    super('steveo-postgres');
+
+    this.prisma = new PrismaClient({
+      datasourceUrl: this.config.datasourceUrl,
+    });
+  }
+
+  /**
+   * Executes the given function under the umbrella of a Postgres transaction.
+   * @param fn
+   */
+  async transaction(fn: (repos: Repositories) => Promise<void>): Promise<void> {
+    this.logger.trace({ msg: `Postgres storage transaction begin` });
+
+    //
+    const opts = {
+      timeout: 20000,
```

Comment:
Timeout should be part of storage config or part of step config?

Reply:
In my opinion it should be part of the storage config as the step config is abstract and has no concept of a transaction
However, I meant to stitch this up and that was just a default I added during development, this has been fixed now)